streams/src/test/java/org/apache/kafka/streams/integration/KStreamAggregationDedupIntegrationTest.java

```diff
@@ -191,39 +194,30 @@ public String apply(Windowed<String> windowedKey, String value) {
 
         startStreams();
 
-        List<KeyValue<String, String>> windowedOutput = receiveMessages(
-            new StringDeserializer(),
-            new StringDeserializer(),
-            10);
-
-        Comparator<KeyValue<String, String>>
-            comparator =
-            new Comparator<KeyValue<String, String>>() {
-                @Override
-                public int compare(final KeyValue<String, String> o1,
-                                   final KeyValue<String, String> o2) {
-                    return KStreamAggregationDedupIntegrationTest.compare(o1, o2);
-                }
-            };
+        Map<String, String> result = tableize(
+            receiveMessages(
+                new StringDeserializer(),
+                new StringDeserializer(),
+                10)
+        );
 
-        Collections.sort(windowedOutput, comparator);
         long firstBatchWindow = firstBatchTimestamp / 500 * 500;
         long secondBatchWindow = secondBatchTimestamp / 500 * 500;
 
-        assertThat(windowedOutput, is(
-            Arrays.asList(
-                new KeyValue<>("A@" + firstBatchWindow, "A"),
-                new KeyValue<>("A@" + secondBatchWindow, "A:A"),
-                new KeyValue<>("B@" + firstBatchWindow, "B"),
-                new KeyValue<>("B@" + secondBatchWindow, "B:B"),
-                new KeyValue<>("C@" + firstBatchWindow, "C"),
-                new KeyValue<>("C@" + secondBatchWindow, "C:C"),
-                new KeyValue<>("D@" + firstBatchWindow, "D"),
-                new KeyValue<>("D@" + secondBatchWindow, "D:D"),
-                new KeyValue<>("E@" + firstBatchWindow, "E"),
-                new KeyValue<>("E@" + secondBatchWindow, "E:E")
-            )
+        final Map<String, String> expected = tableize(Arrays.asList(
+            new KeyValue<>("A@" + firstBatchWindow, "A"),
+            new KeyValue<>("A@" + secondBatchWindow, "A:A"),
+            new KeyValue<>("B@" + firstBatchWindow, "B"),
+            new KeyValue<>("B@" + secondBatchWindow, "B:B"),
+            new KeyValue<>("C@" + firstBatchWindow, "C"),
+            new KeyValue<>("C@" + secondBatchWindow, "C:C"),
+            new KeyValue<>("D@" + firstBatchWindow, "D"),
+            new KeyValue<>("D@" + secondBatchWindow, "D:D"),
+            new KeyValue<>("E@" + firstBatchWindow, "E"),
+            new KeyValue<>("E@" + secondBatchWindow, "E:E")
         ));
+
+        assertEquals(expected, result);
```

Comment:
FWIW I ran locally 25 times and didn't get any failures

Reply:
Oh, it doesn't fail now that I "fixed" the tests. I couldn't figure out why I had to change the tests at all, though.

Reply:
Not sure why you need to drop the ordering check by tableize it? If we cannot find a valid reason we should think twice before changing the test, as it may well expose a regression.

Reply:
My sentiments exactly. I couldn't figure out why the tests started failing, but I'll look at it harder before we think about merging this.

Reply:
What is the new output you see before you change the test?

Reply:
The output contains all the intermediate aggregation states.

Reply:
lol. Upon second glance, this integration test is specifically to verify that this doesn't happen! It's in the name (facepalm).